gravitee-am-service/src/main/java/io/gravitee/am/service/impl/AbstractUserService.java

```diff
@@ -180,10 +182,13 @@ public Maybe<User> findByExternalIdAndSource(ReferenceType referenceType, String
     @Override
     public Single<User> create(ReferenceType referenceType, String referenceId, NewUser newUser) {
         LOGGER.debug("Create a new user {} for {} {}", newUser, referenceType, referenceId);
+        if (StringUtils.isBlank(newUser.getUsername())) {
+            return Single.error(new UserInvalidException("Field [username] is required"));
+        }
```

Comment:
**nitpick (non-blocking)**: I understand you are doing this check to avoid a useless DB request but this check is done into the `create` method line 233. Maybe we can remove this one :thinking:  wdyt ?